app/CocoaPods/CPSplashRecentDocumentsController.m

```diff
@@ -0,0 +1,86 @@
+//
+//  CPSplashRecentDocumentsController.m
+//  CocoaPods
+//
+//  Created by Orta Therox on 05/09/2015.
+//  Copyright (c) 2015 CocoaPods. All rights reserved.
+//
+
+#import "CPSplashRecentDocumentsController.h"
+
+
+@interface _CPSplashDocument : NSObject <NSCopying>
+@property (nonatomic, copy) NSString *name;
+@property (nonatomic, copy) NSURL *podfileURL;
+@property (nonatomic, copy) NSImage *image;
+@property (nonatomic, copy) NSString *folderPath;
+@end
+
+@implementation _CPSplashDocument
+
+// Needed for using Cocoa Bindings
+
+- (instancetype)copyWithZone:(NSZone *)zone
+{
+  _CPSplashDocument *copy = [[[self class] alloc] init];
+  if (copy) {
+    [copy setName:[self.name copyWithZone:zone]];
+    [copy setPodfileURL:[self.podfileURL copyWithZone:zone]];
+    [copy setImage:[self.image copyWithZone:zone]];
+    [copy setFolderPath:[self.folderPath copyWithZone:zone]];
+  }
+  
+  return copy;
+}
+
+@end
+
+@implementation CPSplashRecentDocumentsController
+
+- (void)awakeFromNib
+{
+  [super awakeFromNib];
+    
+  [self.tableView setTarget:self];
+  [self.tableView setDoubleAction:@selector(didDoubleTapOnRecentItem:)];
+  [self setupRecentDocuments];
+}
+
+- (void)setupRecentDocuments
+{
+  NSDocumentController *controller = [NSDocumentController sharedDocumentController];
+  NSMutableArray *documents = [NSMutableArray arrayWithCapacity:controller.recentDocumentURLs.count];
+  for (NSURL *url in controller.recentDocumentURLs) {
+    [documents addObject:[self projectDetailsAtURL:url]];
+  }
+  
+  self.recentDocuments = documents;
+}
+
+- (_CPSplashDocument *)projectDetailsAtURL:(NSURL *)url
+{
+  NSFileManager *fileManager = [NSFileManager defaultManager];
+  NSArray *dirFiles = [fileManager contentsOfDirectoryAtURL:[url URLByDeletingLastPathComponent] includingPropertiesForKeys:nil options:NSDirectoryEnumerationSkipsSubdirectoryDescendants error:nil];
+  NSURL *workspaceURL = [[dirFiles filteredArrayUsingPredicate:[NSPredicate predicateWithFormat:@"self.absoluteString ENDSWITH '.xcworkspace/'"]] firstObject];
+  NSURL *projectURL = [[dirFiles filteredArrayUsingPredicate:[NSPredicate predicateWithFormat:@"self.absoluteString ENDSWITH '.xcodeproj/'"]] firstObject];
+  NSURL *bestURL = workspaceURL ?: projectURL ?: url;
+  
+  _CPSplashDocument *document = [_CPSplashDocument new];
+  document.name = [bestURL lastPathComponent];
+  document.folderPath = [[bestURL pathComponents] componentsJoinedByString:@"/"];
+  document.image = [[NSWorkspace sharedWorkspace] iconForFile:bestURL.absoluteString];
+  document.podfileURL = url;
+  return document;
+}
+
+- (void)didDoubleTapOnRecentItem:(NSTableView *)sender {
+  NSInteger row = [sender clickedRow];
+  _CPSplashDocument *item = self.recentDocuments[row];
```

Comment:
Remember the clickedRow might be `-1` in cases where the user didn't double tap on a row. We should first check that the row is valid before pulling it out of recent documents (and crashing due to index out of range).

```objc
NSInteger row = [sender clickedRow];

if (row != -1) {
  _CPSplashDocument *item = self.recentDocuments[row];
  ...
}
```

From the `clickedRow` documentation:

> This property contains the index of the row that the user clicked. The value is -1 when the user clicks in an area of the table view that is not occupied by table rows.